onnxruntime/core/providers/cpu/nn/regex_full_match.cc

```diff
@@ -0,0 +1,36 @@
+#include "regex_full_match.h"
+#include "core/common/common.h"
+
+namespace onnxruntime {
+ONNX_CPU_OPERATOR_KERNEL(
+    RegexFullMatch,
+    20,
+    KernelDefBuilder()
+        .TypeConstraint("T1", DataTypeImpl::GetTensorType<std::string>())
+        .TypeConstraint("T2", DataTypeImpl::GetTensorType<bool>()),
+    RegexFullMatch);
+
+RegexFullMatch::RegexFullMatch(const OpKernelInfo& info) : OpKernel(info) {
+  ORT_ENFORCE(info.GetAttr<std::string>("pattern", &pattern_).IsOK());
+  ORT_ENFORCE(RE2(pattern_).ok(), "Invalid pattern: ", pattern_);
+}
+
+Status RegexFullMatch::Compute(OpKernelContext* context) const {
+  RE2 re(pattern_);
+  const auto* input_tensor = context->Input<Tensor>(0);
+  if (nullptr == input_tensor) {
+    return Status(common::ONNXRUNTIME, common::FAIL, "Input count mismatch");
+  }
+  auto* output_tensor = context->Output(0, input_tensor->Shape());
+  if (nullptr == output_tensor) {
```

Comment:
If this fails, then the index is wrong, so it must be `ORT_ENFORCE` or assert. However, in practice we do not do that.